test/ui/fixtures/ui_fixtures.py

```
@@ -49,6 +49,84 @@ def is_container_healthy() -> bool:
+    The compose healthcheck only covers Selenium Grid (:4444); code-server
+    may still be starting on :8080 when the health check passes.  We probe
+    from *inside* the container with ``podman exec … curl`` because
+    code-server binds to 127.0.0.1 (container-local loopback) which is not
+    reachable from the host even with a port mapping.
+    """
+    url = "http://127.0.0.1:8080/"
+    deadline = time.time() + timeout_sec
+    attempt = 0
+    while time.time() < deadline:
+        attempt += 1
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)